File: recipe-app-frontend/src/app/services/rating.service.ts
import { RatingDTO } from './../models/rating-model';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class RatingService {

  readonly baseUrl: string = "http://localhost:8080/api/v1/ratings"

  constructor(
    private http: HttpClient
  ) { }

  getByUserId(userId: number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/" + userId);
  }

  checkIfPreviouslyRated(recipeId: number, userId:number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/" + recipeId + "/" + userId)
  }

  getAverageRatingForRecipe(recipeId: any): Observable<any> {
    return this.http.get<any>(this.baseUrl +"/recipe/" + recipeId)
  }

  getTop10Recipes(): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/top10recipes")
  }

  getTop10RecipesForUser(userId: number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/top10recipes/" + userId)
  }

  rateRecipe(ratingDTO: RatingDTO): Observable<any> {
    return this.http.put(`${this.baseUrl}`, ratingDTO)
  }

  

}

File: rating-service/src/main/java/com/ironhack/ratingservice/services/RatingService.java
package com.ironhack.ratingservice.services;

import com.ironhack.ratingservice.dao.Rating;
import com.ironhack.ratingservice.dto.RatingDTO;
import com.ironhack.ratingservice.dto.RecipeDTO;

import java.util.List;

public interface RatingService {

    List<RecipeDTO> findByUserId(Long userId);

    Double getAverageRatingForRecipe(Long recipeId);

    List<RecipeDTO> getTop10Recipes(int i);

    List<RecipeDTO> getTop10RecipesForUser(Long userId);

    boolean userPreviouslyRatedRecipe(Long recipeId, Long userId);

    Rating rateRecipe(RatingDTO ratingDTO);
}
